gocron.go

```diff
@@ -115,20 +115,22 @@ func (j *Job) run() (result []reflect.Value, err error) {
 		}()
 	}
 
-	f := reflect.ValueOf(j.funcs[j.jobFunc])
-	params := j.fparams[j.jobFunc]
+	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
+	j.lastRun = time.Now()
```

Comment:
this undoes https://github.com/jasonlvhit/gocron/pull/116 - I believe we could put it  before calling callJobFuncWithParams? What do you think?

Reply:
Sure, I will move it before the function call. Yet, the fact that all tests ran just fine gives me an uneasy feeling.

cc @aleybovich Could you perhaps think of a unit test that ensures the correct functioning of #116?